### PR TITLE
core: Clipboard component

### PR DESCRIPTION
**Pull request by iBelieve**
Provides access to the current text on the clipboard.

This was used in the Terminal app from Papyros [here](https://github.com/papyros/terminal-app/blob/develop/src/qml/main.qml#L44) and [here](https://github.com/papyros/terminal-app/blob/develop/src/qml/SudoWarningDialog.qml#L41) to warn the user when pasting a command that uses `sudo`.

### CODE REVIEW

File: ui/Icon.qml [hunk 1]
```diff
@@ -103,7 +103,7 @@ Item {
                                 ? name
                                 : name.indexOf("/") !== -1 ? "icon://" + name
                                                            : "image://desktoptheme/" + name
-                                : undefined
+                                : ""
```
Comment (plfiorini) [inline ui/Icon.qml:106]: looks like a change for another commit

File: core/clipboard.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include <QClipboard>
+
+class Clipboard : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY(QString text READ text WRITE setText NOTIFY textChanged)
+
+public:
+    explicit Clipboard(QObject *parent = 0);
```
Comment (plfiorini) [inline core/clipboard.h]: nullptr instead of 0

Comment (iBelieve) [inline core/clipboard.h]: Fixed.

File: ui/Icon.qml (hunk 1 above)
Comment (iBelieve) [inline ui/Icon.qml:106]: Split into separate commit.
